### PR TITLE
feat: add support for doris datasource

### PR DESCRIPTION
**Pull request by chyueyi**
### Describe your changes:

feat <issue-12344>: add support for doris datasource

Apache doris is a very popular mpp database, we use doris as our datawarehouse,so I worked on doris datasource ingestion.

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by harshach** [comment]
hi @chyueyi thanks for your contribution. 
@pmbrull for review

File: ingestion/src/metadata/ingestion/source/database/doris/metadata.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+        return description[0]
+
+    def _get_columns(self, table_name, schema=None):
+        """
+        Overriding the dialect method to add raw_data_type in response
+        """
+
+        table_columns = []
+        primary_columns = []
+        # row schema: Field, Type, Collation, Null, Key, Default, Extra, Privileges, Comment
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/database/doris/metadata.py:202]: Can we add a pedantic model to have a more explicit schema validation and access for the row?

**Comment by TeddyCr** [comment]
Thanks @chyueyi for the PR looks great. I left a comment. Could you also add the python dependency in the `pyproject.toml` for `pydoris` package.

**Comment by chyueyi** [comment]
> Thanks @chyueyi for the PR looks great. I left a comment. Could you also add the python dependency in the `pyproject.toml` for `pydoris` package.

added it👌

File: ingestion/src/metadata/ingestion/source/database/doris/metadata.py (hunk 1 above)
Comment (chyueyi) [reply ingestion/src/metadata/ingestion/source/database/doris/metadata.py:202]: Sorry,I don't know how to do that. Could you please give me a example @TeddyCr

**Review by TeddyCr — APPROVED** [review]

**Comment by Sachin-chaurasiya** [comment]
Hello @chyueyi , thanks for the PR, can you add the doc for the source [here](https://github.com/open-metadata/OpenMetadata/tree/main/openmetadata-ui/src/main/resources/ui/public/locales/en-US/Database)?

Let us know if you need any help.

**Review by chirag-madlani — APPROVED** [review]

**Comment by TeddyCr** [comment]
> Hello @chyueyi , thanks for the PR, can you add the doc for the source [here](https://github.com/open-metadata/OpenMetadata/tree/main/openmetadata-ui/src/main/resources/ui/public/locales/en-US/Database)?
> 
> Also, please attach the screenshot or recording for the UI.
> 
> Feel free to let us know if you need any help.

@chyueyi regarding this comment could you open it in a new PR?

**Comment by chyueyi** [comment]
> > Hello @chyueyi , thanks for the PR, can you add the doc for the source [here](https://github.com/open-metadata/OpenMetadata/tree/main/openmetadata-ui/src/main/resources/ui/public/locales/en-US/Database)?
> > Also, please attach the screenshot or recording for the UI.
> > Feel free to let us know if you need any help.
> 
> @chyueyi regarding this comment could you open it in a new PR?

OK

**Comment by sureshms** [comment]
@TeddyCr and @chyueyi, can some of these issues be closed? https://github.com/open-metadata/OpenMetadata/issues?q=is%3Aissue+is%3Aopen+doris

**Pull request merged**